core/math/quaternion.cpp

```diff
@@ -189,16 +195,70 @@ Quaternion Quaternion::slerpni(const Quaternion &p_to, const real_t &p_weight) c
 			invFactor * from.w + newFactor * p_to.w);
 }
 
-Quaternion Quaternion::cubic_slerp(const Quaternion &p_b, const Quaternion &p_pre_a, const Quaternion &p_post_b, const real_t &p_weight) const {
-#ifdef MATH_CHECKS
-	ERR_FAIL_COND_V_MSG(!is_normalized(), Quaternion(), "The start quaternion must be normalized.");
-	ERR_FAIL_COND_V_MSG(!p_b.is_normalized(), Quaternion(), "The end quaternion must be normalized.");
+static real_t cubic_interpolate_real(const real_t p_pre_a, const real_t p_a, const real_t p_b, const real_t p_post_b, real_t p_c) {
+	// This is cloned straight from animation.cpp
+
+	real_t p0 = p_pre_a;
+	real_t p1 = p_a;
+	real_t p2 = p_b;
+	real_t p3 = p_post_b;
+
+	real_t t = p_c;
+	real_t t2 = t * t;
+	real_t t3 = t2 * t;
+
+	return 0.5f *
```

Comment:
I remember two attempts to consolidate the math equations. Can we check where they went and see if we're duplicating.

Reply:
This was indeed implemented in Math::cubic_interpolate (and animation was also using it to interpolate reals) so i changed this to use it instead. https://github.com/godotengine/godot/pull/63287/commits/4b425786c48c255be7c24200ff84eb475a3ac262